### ISSUE TITLE
(aws-lambda-python): Asset bundling fails on Windows

### ISSUE THREAD

**Issue by gshpychka** [aws/aws-cdk#18861]
### What is the problem?

`aws-lambda-python` does not work, bundling fails.

### Reproduction Steps

Use the `PythonFunction` construct. Peform a synth.

Code excerpt:

```
        lambda_python.PythonFunction(
            self,
            "my_func",
            entry=os.path.join("src", "my_lambda"),
            handler="handler",
            index="my_index.py",
            runtime=typing.cast(lambda_.Runtime, lambda_.Runtime.PYTHON_3_9)
        )
```

### What did you expect to happen?

The lambda is bundled successfully.

### What actually happened?

Error during synth:
`cp: cannot copy a directory, '/asset_input/', into itself, 'asset-output'.`
`Error: Failed to bundle asset ...`

### CDK CLI Version

2.10.0

### Framework Version

2.10.0

### Node.js Version

16.13.0

### OS

Windows

### Language

Python

### Other information

The same works on Linux somehow. This is related - https://github.com/aws/aws-cdk/issues/18301

Pinning `aws-cdk.aws-lambda-python-alpha==2.3.0a0` solves the issue.

**Comment by corymhall** [comment]
@gshpychka I'm having a hard time reproducing this since I don't have a windows machine. The weird part is that the behavior should not be any different between linux and windows because the bundling is happening in a linux docker container. Can you try reproducing this on a fresh project with just the minimum required dependencies?

**Comment by gbisaga** [comment]
I am having the same problem with

    "@aws-cdk/aws-lambda-python-alpha": "^2.13.0-alpha.0",
    "aws-cdk-lib": "^2.13.0",

Here's a sample minimal project @corymhall :

https://github.com/gbisaga/aws-cdk-issues-18861

**Comment by vduits** [comment]
@gbisaga 
It might be worth re-opening in another issue as the author's location likely means they are occupied with other things and leave them unable to respond and they don't seem to reopen it. It is how they addressed it the first time as well after #18301.

You can still bypass this issue by using v 2.3.0-alpha.0. I still got that working with v2.14 of the aws-cdk-lib without it introducing problems. But I would imagine it will on the long run.

**Comment by corymhall** [comment]
@gbisaga thanks for the sample project! This one really has me stumped. My hunch is that it is something with the way Docker handles volumes on Windows, but I might need some help running down a root cause.

**Comment by gbisaga** [comment]
Of course! Please let me know if you would like me to try anything. I'm not that familiar with the internals of CDK, but I would love to learn more about it! Looking at the timing, I'm wondering if it has anything to do with the recent Docker Desktop changes. I have not experienced any other problems with it, but that might explain both why it doesn't have problems on linux and also why it just started happening now.

**Comment by corymhall** [comment]
@gbisaga can you try running a test to see if it throws the same error?

```
docker run --rm -v /tmp/copy-test:/asset-input:delegated -v /tmp/copy-test-output:/asset-output:delegated -w /asset-input public.ecr.aws/sam/build-python3.9 cp -rT /asset-input/ /asset-output
```

You'll need to replace the volume source paths to windows paths so you'll end up with something like (depending on where you create the directories)

```
docker run --rm -v C:\\copy-test:/asset-input:delegated -v C:\\copy-test-output:/asset-output:delegated -w /asset-input public.ecr.aws/sam/build-python3.9 cp -rT /asset-input/ /asset-output
```

I created two directories `/tmp/copy-test` and `/tmp/copy-test-output`. I cloned your test repository into `copy-test` so I ended up with 

/tmp/copy-test
```
/tmp/copy-test
└── aws-cdk-issues-18861
```

And after running the test I had

```
/tmp/copy-test-output
└── aws-cdk-issues-18861
```

**Comment by kichik** [comment]
I have also been having this issue for a while now. I used `procmon.exe` to get the Docker command that CDK runs and it's:

```
"C:\Program Files\Docker\Docker\resources\bin\com.docker.cli.exe" run --rm -u 1000:1000 -v C:\...\my-proj\my-lambda:/
asset-input:delegated -v C:\...\my-proj\cdk.out\asset.94afcae579c0c4746c303352f6aae43209f50ce029812eb3c005bdd3fb8519bb:/asset-output:delegated -w /asset-input cdk-6bbe7af8159970c0254f59e44d4d98e5cd784e14832228c913df8ea
651d8b90d bash -c "python -m pip install -r requirements.txt -t \asset-output && cp -rT /asset-input/ \asset-output"
```

The issue seems to be the last two `\asset-output`. Once I change those to `/asset-output` everything works again. I'm not sure why CDK chooses to use Windows style slashes inside Docker.

**Comment by corymhall** [comment]
Leaving this open until we can confirm the fix works.

**Comment by kichik** [comment]
Can I help confirm before the next release somehow?

**Comment by corymhall** [comment]
@kichik can you try with the latest version? The fix should be in `1.148.0` and `2.17.0`(just released).
